packages/common/src/services/audius-backend/solana.ts

```diff
@@ -107,66 +125,56 @@ export const createUserBankIfNeeded = async (
     ethAddress ?? audiusLibs.Account!.getCurrentUser()?.wallet
 
   if (!recipientEthAddress) {
-    console.error(
-      "createUserBankIfNeeded: Unexpectedly couldn't get recipient eth address"
+    throw new Error(
+      `createUserBankIfNeeded: Unexpectedly couldn't get recipient eth address`
     )
-    return
   }
 
   try {
-    const res = await audiusLibs.solanaWeb3Manager!.createUserBankIfNeeded({
-      feePayerOverride,
-      ethAddress: recipientEthAddress,
-      mint
-    })
+    const res: CreateUserBankIfNeededResult =
+      await audiusLibs.solanaWeb3Manager!.createUserBankIfNeeded({
+        feePayerOverride,
+        ethAddress: recipientEthAddress,
+        mint
+      })
 
-    // If it already existed, return early
-    if ('didExist' in res && res.didExist) {
-      console.log('Userbank already exists')
-      return
+    if (isCreateUserBankIfNeededError(res)) {
+      // Will catch and log below
+      throw res
     }
 
-    // Otherwise we must have tried to create one
-    console.info(`Userbank doesn't exist, attempted to create...`)
-
-    recordAnalytics({
-      eventName: Name.CREATE_USER_BANK_REQUEST,
-      properties: { mint, recipientEthAddress }
-    })
+    // If it already existed, return early
+    if (res.didExist) {
+      console.log('Userbank already exists')
+    } else {
+      // Otherwise we must have tried to create one
+      console.info(`Userbank doesn't exist, attempted to create...`)
 
-    // Handle error case
-    if ('error' in res) {
-      console.error(
-        `Failed to create userbank, with err: ${res.error}, ${res.errorCode}`
-      )
       recordAnalytics({
-        eventName: Name.CREATE_USER_BANK_FAILURE,
-        properties: {
-          mint,
-          recipientEthAddress,
-          errorCode: res.errorCode,
-          error: (res.error as any).toString()
-        }
+        eventName: Name.CREATE_USER_BANK_REQUEST,
+        properties: { mint, recipientEthAddress }
       })
-      return
     }
 
-    // Handle success case
-    console.log(`Successfully created userbank!`)
     recordAnalytics({
       eventName: Name.CREATE_USER_BANK_SUCCESS,
       properties: { mint, recipientEthAddress }
     })
-  } catch (err) {
+    return res.userbank
+  } catch (err: any) {
+    // Catching error here for analytics purposes
+    const errorMessage = 'error' in err ? err.error : (err as any).toString()
+    const errorCode = 'errorCode' in err ? err.errorCode : undefined
     recordAnalytics({
       eventName: Name.CREATE_USER_BANK_FAILURE,
       properties: {
         mint,
         recipientEthAddress,
-        errorMessage: (err as any).toString()
+        errorCode,
+        errorMessage
       }
     })
-    console.error(`Failed to create userbank, with err: ${err}`)
+    throw new Error(`Failed to create user bank: ${errorMessage}`)
```

Comment:
sure we want to throw here? would that brick the app if we forget to catch it downstream?

Reply:
Yeah I debated this. We only use this code in a few places. In all of them, any of the failures that would lead here would put us in a state where continuing isn't feasible (the userbank doesn't exist, so we can't move on with the saga that wants to use it).

I'm happy to remove this for now, but I think it will result in strange behavior at best where calling code isn't handling these error cases and we end up throwing in a different place further down the line?

Reply:
cc @dylanjeffers @rickyrombo @sliptype Do we have any conventions around error handling in sagas? Is it more consistent here to just return nothing and let calling code detect the absence of a value? Or should we throw errors and let calling code try/catch?